### PR TITLE
migrate to mocha, split out plugins, mapparams defaults to false

### PR DESCRIPTION
**Pull request by DonutEspresso**
WIP

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	$(JSCS) $(SRCS) --fix
+
+
+.PHONY: prepush
+prepush: node_modules lint codestyle test nsp
+
+
+.PHONY: test
+test: node_modules
+	$(MOCHA) -R spec --full-trace
```
Comment (micahr) [inline Makefile]: whoa, mocha, i dunno... :)

Comment (DonutEspresso) [inline Makefile]: well, errors and clients already use it, so @yunong will have to get used to it ;)

**Comment by DonutEspresso** [comment]
@yunong @micahr plugins broken out! 

All functionality is intact, added more unit tests for queryParser and bodyParser. Also made mapParams: false the default behavior.

**Comment by micahr** [comment]
Looks great, thanks for breaking this out @DonutEspresso. It would be great to get this in for 5.0

**Comment by yunong** [comment]
@DonutEspresso looks good. If we're going this route, let's move all plugins out of server, even CORS. It's fine for the server to pull it in as a dependency. Also, when you're moving the files over to new repos, please save the git logs for the files so we're not starting anew.

**Comment by DonutEspresso** [comment]
Not sure what to do with CORS - the tests require restify so pulling CORS out would result in a circular dependency for tests (or we keep CORS tests in core). We also have existing PRs open against CORS. 

The git history is complex, let me see what I can do.

**Pull request closed without merging**